compiler/rustc_hir_analysis/src/collect/item_bounds.rs

```diff
@@ -78,6 +78,8 @@ pub(super) fn explicit_item_bounds(
 ) -> &'_ [(ty::Predicate<'_>, Span)] {
     // If the def_id is about an RPITIT, delegate explicit_item_bounds to the opaque_def_id that
     // generated the synthesized associate type.
+    // FIXME(-Zlower-impl-trait-in-trait-to-assoc-ty): This is not correct for bounds with nested
+    // RPITITs.
```

Comment:
What did you mean here exactly?

Reply:
Maybe I misled myself, will probably just drop this comment and think about it more.